Explain that props are updated on navigation (#6915)
withastro/docs: src/content/docs/en/guides/view-transitions.mdx

You can also place the directive on an [Astro island](/en/concepts/islands/) (a UI framework component with a [`client:` directive](/en/reference/directives-reference/#client-directives)). If that component exists on the next page, the island from the old page **with its current state** will continue to be displayed, instead of replacing it with the island from the new page.

> In the example below, the component's `count` internal state will not be reset when navigating back and forth across pages that contain the `<Counter />` component with the `transition:persist` attribute.

Reviewer: ```suggestion
In the example below, the component's `initialCount` internal state will not be reset when navigating back and forth across pages that contain the `<Counter />` component with the `transition:persist` attribute.
```

<video controls="" autoplay="" transition:persist="media-player">
```

#### `transition:persist-props`
<p><Since v="4.5.0" /></p>

> **Default**: `false`

Reviewer: ```suggestion
```

This allows you to control whether or not an island's props should be persisted upon navigation.

By default, when you add `transition:persist` to an island, the state is retained upon navigation, but your component will re-render with new props. This is useful, for example, when a component receives page-specific props such as the current page's `title`.

> You can override this behavior with `transition:persist-props`. Adding this directive will keep an island's existing props (not re-render with new values) in addition to maintaining its existing state.

Reviewer: ``````suggestion
You can override this behavior with `transition:persist-props`. Adding this directive will keep an island's existing props (not re-render with new values) in addition to maintaining its existing state.

```astro
<ProductListing transition:persist-props=true products={products} />
```
``````
Thanks @matthewp ! Wording looks fine to me, but if they do have to set this to `true`, then we should probably show an example of that. (If they *don't* have to set it to true, and just merely adding it adds it as true, then I would choose to remove the "Default: false" from above.)

Author: They don't have to set it to true. It works like a boolean attribute where the presence of the attribute makes it be true. I can see how the "defaults to false" part might be confusing, even if its technically accurate, so we can remove if you think that's best.

Reviewer: Great, then I'll remove the default line and we're good to go!

<video controls="" autoplay="" transition:persist="media-player">
```

> When you add `transition:persist` to an island, the state is retained upon navigation, but any props that change will be carried over to the new page, causing your component to re-render with new props, but the existing state. You can control this behavior with the `transition:persist-props` directive. Adding this directive will keep the existing props and not copy over new values.

Reviewer: ```suggestion
#### `transition:persist-props`
<p><Since v="4.5.0" /></p>
**Default**: `false`

This allows you to control whether or not an island's props should be persisted upon navigation.

By default, when you add `transition:persist` to an island, the state is retained upon navigation, but your component will re-render with new props. This is useful, for example, when a component receives page-specific props such as the current page's `title`.

You can override this behavior with `transition:persist-props="true"`. Adding this directive will keep an island's existing props (not re-render with new values) in addition to maintaining its existing state.
```
Here's my suggestion, so we can mention when this was introduced too.

Author: You don't need the `="true"` part but otherwise very good! I can't edit the suggestion so I think i'll commit and then change.
